drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1187,6 +1189,103 @@ static int eth_initialize(const struct device *dev)
 	return 0;
 }
 
+static uint32_t reverse(uint32_t val)
+{
+	uint32_t res = 0;
+	int i;
+
+	for (i = 0; i < 32; i++) {
+		if (val & (1 << i)) {
+			res |= 1 << (31 - i);
+		}
+	}
+
+	return res;
+}
+
+static void net_if_mcast_cb(struct net_if *iface,
+			    const struct net_addr *addr,
+			    bool is_joined)
+{
+	ARG_UNUSED(addr);
+	ARG_UNUSED(is_joined);
+
+	const struct device *dev;
+	struct eth_stm32_hal_dev_data *dev_data;
+	ETH_HandleTypeDef *heth;
+#if defined(CONFIG_NET_NATIVE_IPV6)
+	struct net_if_ipv6 *ipv6;
+#endif
+#if defined(CONFIG_NET_NATIVE_IPV4)
+	struct net_if_ipv4 *ipv4;
+#endif
+	struct net_eth_addr mac_addr;
+	uint32_t crc;
+	uint32_t hash_table[2];
+	uint32_t hash_index;
+	int i;
+
+	__ASSERT_NO_MSG(iface != NULL);
+
+	dev = net_if_get_device(iface);
+	__ASSERT_NO_MSG(dev != NULL);
+
+	dev_data = DEV_DATA(dev);
```

Comment:
`DEV_DATA` usage is not welcome anymore. See https://github.com/zephyrproject-rtos/zephyr/pull/41918

Reply:
Thanks, this commit was in the original. In my modified commit, I removed the use of DEV_DATA: https://github.com/zephyrproject-rtos/zephyr/pull/53850/commits/d6f517371b25ab063459decc691407dc383711ad#diff-0d2e5611dcfdafb92af335bba5c381139ed9ce88182b5fe03d9655f7984725bcR1241